### PR TITLE
8333590: UnmodifiableHeaders.toString() returns a value that represents empty headers

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which proposes to fix the issue noted in https://bugs.openjdk.org/browse/JDK-8333590?

As noted in that issue the `sun.net.httpserver.UnmodifiableHeaders`, an internal class, within the `jdk.httpserver` module doesn't override the `toString()` method. `UnmodifiableHeaders` is of type `com.sun.net.httpserver.Headers` which is a public API in the `jdk.httpserver` module.

When applications call `Headers.toString()`, if the instance is of type `UnmodifiableHeaders`, then the returned `String` representation shows empty headers, even when it isn't empty. The commit in this PR fixes the issue by implementing `toString()` in the `UnmodifiableHeaders` to return the correct representation of the headers.

An existing jtreg test has been updated to reproduce this issue and verify the fix.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333590](https://bugs.openjdk.org/browse/JDK-8333590): UnmodifiableHeaders.toString() returns a value that represents empty headers (**Bug** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)
 * [Michael McMahon](https://openjdk.org/census#michaelm) (@Michael-Mc-Mahon - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19555/head:pull/19555` \
`$ git checkout pull/19555`

Update a local copy of the PR: \
`$ git checkout pull/19555` \
`$ git pull https://git.openjdk.org/jdk.git pull/19555/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19555`

View PR using the GUI difftool: \
`$ git pr show -t 19555`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19555.diff">https://git.openjdk.org/jdk/pull/19555.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19555#issuecomment-2149484706)

### CODE REVIEW

File: src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java [hunk 1]
```diff
@@ -32,73 +32,96 @@
...
 
+    @Override
     public boolean equals(Object o) {return headers.equals(o);}
 
+    @Override
     public int hashCode() {return headers.hashCode();}
+
+    @Override
+    public String toString() {
+        return this.headers.toString();
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java:125]:
```suggestion
        return headers.toString();
```
to keep with the style of other methods in this class

**Review by dfuch — APPROVED** [review]
LGTM

**Review by Michael-Mc-Mahon — APPROVED** [review]
Looks fine.

**Comment by jaikiran** [comment]
Thank you Daniel and Michael for the reviews. CI tier tests came back fine. I'll go ahead with the integration shortly.

**Pull request closed without merging**